.github/workflows/stacks-signer-tests.yml

```diff
@@ -0,0 +1,33 @@
+## Run Stacks Signer Unit Tests
+name: Tests::Stacks Signer Unit Tests
+
+on:
+  workflow_call:
+
+env:
+  BITCOIND_TEST: 0
+  RUST_BACKTRACE: full
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 3
+  TEST_TIMEOUT: 30
+
+concurrency:
+  group: stacks-core-tests-${{ github.head_ref || github.ref || github.run_id }}
+  # Only cancel in progress if this is for a PR
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  unit-tests:
+    name: Unit Tests
+    runs-on: ubuntu-latest
+    steps:
+      ## Setup test environment
+      - name: Setup Test Environment
+        id: setup_tests
+        uses: stacks-network/actions/stacks-core/testenv@main
+        with:
+          btc-version: "25.0"
+
+      # Run unit tests
+      - name: Run Unit Tests
+        run: cargo test -p stacks-signer -- --nocapture --test-threads 1
```

Comment:
Do these have to run on only one thread or could they run in parallel?

Reply:
it may also make sense to use the nextest archive vs rebuilding the signer binary solely for this test. 
depending on what we're testing here, this may be able to copy something like this: https://github.com/stacks-network/stacks-core/blob/master/.github/workflows/bitcoin-tests.yml#L130-L138

we can also adjust the composite that runs the test from the archive